airbyte-webapp/src/views/Connection/ConnectionForm/formConfigHelpers.ts

```diff
@@ -0,0 +1,88 @@
+import isEmpty from "lodash/isEmpty";
+
+import {
+  AirbyteStreamConfiguration,
+  DestinationSyncMode,
+  SyncMode,
+  SyncSchemaStream,
+} from "../../../core/domain/catalog";
+
+const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
+  if (streamNode.stream.defaultCursorField.length) {
+    return streamNode.stream.defaultCursorField;
+  }
+  return streamNode.config.cursorField;
+};
+
+const verifySupportedSyncModes = (streamNode: SyncSchemaStream): SyncSchemaStream => {
+  const {
+    stream: { supportedSyncModes },
+  } = streamNode;
+
+  if (!isEmpty(supportedSyncModes)) return streamNode;
```

Comment:
```suggestion
  if (!supportedSyncModes.length) {
    return streamNode;
  }
```
isEmpty is overkill given we know this is an array.

Reply:
Ok, will fix it